### ISSUE TITLE
Combine Vanilla JS

### ISSUE THREAD

**Issue by IndraGunawan** [symfony/webpack-encore#149]
is there a way to combining vanilla js?

something like https://laravel.com/docs/5.4/mix#vanilla-js

**Comment by stof** [comment]
Nope. webpack-encore is a builder for webpack configuration object. It does not implement any processing on its own.
There is no plan to change this. If you just want to concatenate files together, use a different tool for that (it could be gulp with gulp-concat for instance)

**Comment by Lyrkan** [comment]
I've seen someone talk about [this Webpack plugin](https://www.npmjs.com/package/webpack-concat-plugin) in another issue (I think it was #136 but the response has been deleted since then).

Never used it myself but based on its description it could fit your needs:

```js
const ConcatPlugin = require('webpack-concat-plugin');

Encore.addPlugin(new ConcatPlugin({
    fileName: 'output.js',
    filesToConcat: [
        './file1.js',
        './file2.js'
    ]
}))
```

**Comment by IndraGunawan** [comment]
@stof thanks for the information.

@Lyrkan thanks. i have already test that webpack plugin,  then the problem is when i concat with hash file name, the filename does not include in `manifest.json` that created by encore.

i think its good if encore can create something like dispatch an event when creating the `manifest.json` cc: @weaverryan 
ps: i am new in js, i don't know it can be implement or not.

**Comment by Lyrkan** [comment]
There is actually another issue about that (it didn't show up when searching for "concat"): #140

Looks like it could be solved after some changes in the Webpack Manifest Plugin.

**Comment by SvetlinStaev** [comment]
Hi, I am running into a similar problem with the concat js file not available in the manifest, I see that the issuer has been closed in the `Webpack Manifest Plugin`, but the file is still not injectable in the `manifest.json`, is there any additional configuration necessary in Encore to do that?

**Comment by stof** [comment]
@weaverryan could it be that custom plugins are added after the manifest plugin (and so files they inject are not known by the manifest plugin) ?

**Comment by Lyrkan** [comment]
@stof That's probably needed but [this PR](https://github.com/danethurber/webpack-manifest-plugin/pull/75) hasn't been released yet anyway (merged 3 days ago).

**Comment by weaverryan** [comment]
There are 2 things going on here:

1) That PR (https://github.com/danethurber/webpack-manifest-plugin/pull/75) was merged into master. There are a few breaking changes that are on master... waiting for a v2 release. So, we're waiting on that.

2) We don't actually use https://github.com/danethurber/webpack-manifest-plugin :). We have a copy of it hardcoded into our repo. There were 2 issues (I believe both fixed on `master`, so in v2) that are well-commented in our version of the file that caused this.

tl;dr When v2 of that library is released, we should be able to start using it again (woo!) and then this issue will be closed.

**Comment by weaverryan** [comment]
Closing. We now use the official webpack-manifest-plugin lib, and you can configure it via `Encore. configureManifestPlugin()`.

Cheers!
